src/rebound.c

```diff
@@ -69,6 +69,13 @@ const char* reb_githash_str = STRINGIFY(GITHASH);             // This line gets
 void reb_step(struct reb_simulation* const r){
     // A 'DKD'-like integrator will do the first 'D' part.
     PROFILING_START()
+    if (r->pre_timestep_modifications){
+        reb_integrator_synchronize(r);
+        r->pre_timestep_modifications(r);
+        r->ri_whfast.recalculate_jacobi_this_timestep = 1;
```

Comment:
Is that a good idea to set these flags here? It's not symmetric. I.e. not what we do for the post_timestep_modifications, where the user has to set the flags manually if desired.